cpp/src/arrow/util/decimal_test.cc

```diff
@@ -1521,6 +1525,10 @@ TEST(Decimal128Test, ReduceScaleAndRound) {
   result = Decimal128("-123750").ReduceScaleBy(2, true);
   ASSERT_OK(result.ToInteger(&out));
   ASSERT_EQ(-1238, out);
+
+  result = Decimal128("-5").ReduceScaleBy(1, true);
+  ASSERT_OK(result.ToInteger(&out));
+  ASSERT_EQ(-1, out);
```

Comment:
Can we also add tests for `Decimal128(0)`?

Reply:
Done